nalgebra-lapack/src/generalized_eigenvalues.rs

```diff
@@ -220,12 +220,13 @@ where
 
         let mut c = 0;
 
+        let epsilon = T::RealField::default_epsilon();
+
         while c < n {
-            if eigenvalues[c].im.abs() > T::RealField::default_epsilon() && c + 1 < n && {
+            if eigenvalues[c].im.abs() > epsilon && c + 1 < n && {
```

Comment:
I missed this the first time around, but at first glance this logic does not sound to me. What does it do? I think this needs a solid comment and/or documentation.

For example, imagine scaling a matrix by `1e20`, then all its eigenvalues get scaled by `1e20` and this logic will behave entirely differently. It seems to me at the *very least* that we'd need to look relative to the maximum absolute eigenvalue, but even then it's not bullet proof.

Reply:
The scaling point is something I've never considered, it's a pretty solid point. 

But in this scenario, LAPACK uses the eigenvalues to signal how to assemble eigenvectors. If the eigenvalue is complex and if the subsequent eigenvalue is the former's complex conjugate, the eigenvector to be assembled has it's real and imaginary parts in two subsequent columns.

Does the scaling edge case apply here seeing that it's more of an assembling problem by signalling than a numeric one? In the documentation, the way to check for realness is by looking at a_j values, this gets mapped to the imaginary value of an eigenvector. While it's interesting for me to look deeper into LAPACK code, was wondering if putting the burden of getting right for the edge case is more on LAPACK to get it right?

In either case, putting in a comment here does make sense

Reply:
I think for me it's not clear why we need to do any kind of "post processing" at all. I think this would be nice to have a comment explaining this. And moreover, I would be very surprised if LAPACK expects you to use approximate comparison if it's signaling something through values - are you sure *exact* comparison is not more appropriate? I just don't have enough context to know what is the appropriate thing to do here.

Reply:
This is fair. I think there is some level of post processing expected but not to the extent I have done which is more a result of not trusting LAPACK which may be unwarranted. The changes reflect what I think LAPACK expects us to do minimally